scripts/run_docking_oe.py

```diff
@@ -0,0 +1,310 @@
+"""
+Script to dock an SDF file of ligands to prepared structures.
+"""
+import argparse
+from glob import glob
+import multiprocessing as mp
+from openeye import oechem
+import os
+import pandas
+import pickle as pkl
+import re
+import shutil
+import sys
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from covid_moonshot_ml.datasets.utils import load_openeye_sdf, save_openeye_sdf
+from covid_moonshot_ml.docking.docking import run_docking_oe
+
+
+def check_results(d):
+    """
+    Check if results exist already so we can skip.
+
+    Parameters
+    ----------
+    d : str
+        Directory
+
+    Returns
+    -------
+    bool
+        Results already exist
+    """
+    if (not os.path.isfile(os.path.join(d, "docked.sdf"))) or (
+        not os.path.isfile(os.path.join(d, "results.pkl"))
+    ):
+        return False
+
+    try:
+        _ = load_openeye_sdf(os.path.join(d, "docked.sdf"))
+    except Exception:
+        return False
+
+    try:
+        _ = pkl.load(open(os.path.join(d, "results.pkl"), "rb"))
+    except Exception:
+        return False
+
+    return True
+
+
+def load_dus(file_base):
+    """
+    Load all present oedu files. If `file_base` is a directory, os.walk will be
+    used to find all .oedu files in the directory. Otherwise, it will be
+    assessed with glob.
+
+    Parameters
+    ----------
+    file_base : str
+        Directory/base filepath for .oedu files.
+
+    Returns
+    -------
+    Dict[str, List[oechem.OEDesignUnit]]
+        Dictionary mapping dataset to list of design units
+    """
+
+    if os.path.isdir(file_base):
+        all_fns = [
+            os.path.join(file_base, fn)
+            for _, _, files in os.walk(file_base)
+            for fn in files
+            if fn[-4:] == "oedu"
+        ]
+    else:
+        all_fns = glob(file_base)
+
+    du_dict = {}
+    re_pat = r"(Mpro-[A-Za-z][0-9]+)_"
+    for fn in all_fns:
+        dataset = re.search(re_pat, fn)
+        if dataset is None:
+            print(f"No Mpro dataset found for {fn}", flush=True)
+            continue
+
+        dataset = dataset.groups()[0]
+        du = oechem.OEDesignUnit()
+        if not oechem.OEReadDesignUnit(fn, du):
+            print(f"Failed to read DesignUnit {fn}", flush=True)
+            continue
+        try:
+            du_dict[dataset].append(du)
+        except KeyError:
+            du_dict[dataset] = [du]
```

Comment:
@kaminow is this what is capturing the prepped du directory name? I think thats why we've dropped the chain information downstream, it's only capturing before the `_`? We can get the original compound_id but we should definitely keep track somewhere if we're getting the, i.e. Mpro-P0060_0A or Mpro-P0060_0B structures.

Reply:
fixed now!